tasks/task_database_cue.c

```diff
@@ -127,231 +217,654 @@ static int64_t get_token(intfstream_t *fd, char *token, uint64_t max_len)
+      {
+         strncpy(lgame_id, &pre_game_id[3], 4);
+         lgame_id[4] = '\0';
+         strcat(game_id, lgame_id);
+         strcat(game_id, rgame_id);
+         cue_append_multi_disc_suffix(game_id, filename);
+         return true;
+      }
+      else
+      {
+         strncpy(game_id, &pre_game_id[3], 4);
+         game_id[4] = '\0';
+         cue_append_multi_disc_suffix(game_id, filename);
+         return true;
+      }
+   }
+   else
+   {
+      string_trim_whitespace(raw_game_id);
+      strcpy(game_id, raw_game_id);
+      return true;
    }
+   return false;
+}
+
+int detect_sat_game(intfstream_t *fd, char *game_id, const char *filename)
+{
+   char hyphen = '-';
+   char raw_game_id[15];
+   char raw_region_id[15];
+   char region_id;
+   char check_prefix_t_hyp[10];
+   char check_prefix_mk_hyp[10];
+   int length;
+   char lgame_id[10];
+   char rgame_id[10];
+   char game_id50[] = "-50";
+
+   /* Load raw serial or quit */
+   if (intfstream_seek(fd, 0x0030, SEEK_SET) < 0)
+      return false;
 
-   tmp = boot_file;
-   *game_id++ = toupper(*tmp++);
-   *game_id++ = toupper(*tmp++);
-   *game_id++ = toupper(*tmp++);
-   *game_id++ = toupper(*tmp++);
-   *game_id++ = '-';
+   if (intfstream_read(fd, raw_game_id, 9) <= 0)
+      return false;
 
-   if (!ISALNUM(*tmp))
-      tmp++;
+   raw_game_id[9] = '\0';
 
-   while (ISALNUM(*tmp))
+   /* Load raw region id or quit */
+   if (intfstream_seek(fd, 0x0050, SEEK_SET) < 0)
+      return false;
+
+   if (intfstream_read(fd, raw_region_id, 1) <= 0)
+      return false;
+
+   raw_region_id[1] = '\0';
+
+   /** Scrub files with bad data and log **/
+   if (raw_game_id[0] == '\0' || raw_game_id[0] == ' ')
    {
-      *game_id++ = *tmp++;
-      if (*tmp == '.')
-         tmp++;
+      RARCH_LOG("[Scanner]: Scrubbing: %s\n", filename);
+      return false;
    }
 
-   *game_id = 0;
+   region_id = raw_region_id[0];
 
-   return 1;
-}
+   string_trim_whitespace(raw_game_id);
 
-int detect_ps1_game(intfstream_t *fd, char *game_id)
-{
-   if (detect_ps1_game_sub(fd, game_id, 0))
-      return 1;
+   /** disect this raw serial into parts **/
+   strncpy(check_prefix_t_hyp, raw_game_id, 2);
+   check_prefix_t_hyp[2] = '\0';
+   strncpy(check_prefix_mk_hyp, raw_game_id, 3);
+   check_prefix_mk_hyp[3] = '\0';
+   length = strlen(raw_game_id);
+   raw_game_id[length] = '\0';
 
-   return detect_ps1_game_sub(fd, game_id, 1);
+   /** redump serials are built differently for each region **/
+   switch (region_id)
+   {
+      case 'U':
+         if (strcmp(check_prefix_mk_hyp, "MK-") == 0)
+         {
+            strncpy(game_id, &raw_game_id[3], length - 3);
+            game_id[length - 3] = '\0';
+            cue_append_multi_disc_suffix(game_id, filename);
+            return true;
+         }
+         else
+         {
+            strcpy(game_id, raw_game_id);
+            cue_append_multi_disc_suffix(game_id, filename);
+            return true;
+         }
```

Comment:
Return true at the end to avoid the look of an unending case.
``` suggestion
            cue_append_multi_disc_suffix(game_id, filename);
         }
         else
         {
            strcpy(game_id, raw_game_id);
            cue_append_multi_disc_suffix(game_id, filename);
         }
         return true;
```